ttnn/cpp/pybind11/pytensor.cpp

```diff
@@ -872,16 +1016,15 @@ void pytensor_module(py::module& m_tensor) {
                           Layout layout,
                           const MemoryConfig& mem_config,
                           const std::optional<Tile>& tile) {
-                auto tensor = detail::convert_python_tensor_to_tt_tensor(python_tensor, data_type, tile);
-                auto layout_tensor = tensor.to(layout);
-                return layout_tensor.to(device, mem_config);
+                return detail::convert_python_tensor_to_tt_tensor(
+                    python_tensor, data_type, layout, tile, mem_config, device, true);
             }),
             py::arg("tensor"),
             py::arg("data_type") = std::nullopt,
-            py::arg("device").noconvert(),
-            py::arg("layout").noconvert(),
-            py::arg("mem_config").noconvert(),
-            py::arg("tile") = std::nullopt,
+            py::arg("device") = nullptr,
+            py::arg("layout").noconvert() = Layout::ROW_MAJOR,
```

Comment:
Why add these defaults? Shouldn't other `py::init<>` bindings take care of cases where layout is not specified?

How should this be reflected in the doc string?

Reply:
without defaults, the pybound method `ttnn.Tensor(py_tensor, ...)` for this option is really unusable, since it forces you to provide all arguments. Also, the `device.noconvert()` means you can't pass `None` if you want this pass to be on host instead.

Reply:
Regarding documentation, I think it's actually wrong to start with, since that code should go to the previous overloaded definition. I will fix it.